contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2135,13 +2142,30 @@ exec_alter_role_cmd(char *query_str, RoleSpec *role)
  * Helper function to generate GRANT on SCHEMA subcommands.
  */
 static List
-*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege)
+*gen_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege, bool is_create_schema)
 {
 	StringInfoData query;
 	List	   *stmt_list;
 	Node	   *stmt;
 	int			expected_stmts = 2;
 	int			i = 0;
+	bool		owner_other_than_dbo = false;
+	char 		*dbname = get_cur_db_name();
+	const char	*dbo_role = get_dbo_role_name(dbname);
+	const char	*db_owner_role = get_db_owner_name(dbname);
+	const char	*schema_owner = GetUserNameFromId(get_owner_of_schema(schema), false);
```

Comment:
Now that I think about this, none of these objects should be `const`. The functions like `get_dbo_role_name`, `get_db_owner_name` etc return both const and palloc'd strings so it can't be const depending upon whether the database is builtin or user defined. This is not correct since in case of user defined database, the function returns palloc'd string but we can't free it due to `const` qualifier. Rather such functions should return palloc'd string even for builtin databases and then caller should always free the returned value.
*Suggestion*: Can take this opportunity to fix few of such functions.

Reply:
Let's take that separately since this JIRA is marked for Vole and we don't have enough time left.